src/pudl/analysis/service_territory.py

```diff
@@ -536,77 +551,145 @@ def plot_all_territories(
+    type=click.Choice(
+        ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], case_sensitive=False
+    ),
+    show_default=True,
+)
+def pudl_service_territories(
+    entity_type: Literal["util", "ba"],
+    dissolve: bool,
+    output_dir: pathlib.Path,
+    limit_by_state: bool,
+    logfile: pathlib.Path,
+    loglevel: str,
+):
+    """Compile historical utility and balancing area service territory geometries.
 
-    Args:
-        argv (list): command line arguments including caller file name.
+    This script produces GeoParquet files describing the historical service territories
+    of utilities and balancing authorities based on data reported in the EIA Form 861
+    and county geometries from the US Census DP1 geodatabase.
 
-    Returns:
-        dict: A dictionary mapping command line arguments to their values.
-    """
-    parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument(
-        "-d",
-        "--dissolve",
-        dest="dissolve",
-        action="store_true",
-        default=False,
-        help="Dissolve county level geometries to utility or balancing authorities",
-    )
-    parser.add_argument(
-        "--logfile",
-        default=None,
-        type=str,
-        help="If specified, write logs to this file.",
-    )
-    parser.add_argument(
-        "--loglevel",
-        help="Set logging level (DEBUG, INFO, WARNING, ERROR, or CRITICAL).",
-        default="INFO",
-    )
-    return parser.parse_args(argv[1:])
+    See: https://geoparquet.org/ for more on the GeoParquet file format.
 
+    Usage examples:
 
-def main():
-    """Compile historical utility and balancing area territories."""
+    pudl_service_territories --entity-type ba --dissolve --limit-by-state
+    pudl_service_territories --entity-type util
+    """
     # Display logged output from the PUDL package:
+    pudl.logging_helpers.configure_root_logger(logfile=logfile, loglevel=loglevel)
 
-    args = parse_command_line(sys.argv)
-    pudl.logging_helpers.configure_root_logger(
-        logfile=args.logfile, loglevel=args.loglevel
+    pudl_engine = sa.create_engine(PudlPaths().pudl_db)
+    # Load the required US Census DP1 county geometry data:
+    dp1_engine = PudlPaths().sqlite_db_uri("censusdp1tract")
+    sql = """
+SELECT
+    geoid10,
+    namelsad10,
+    dp0010001,
+    shape AS geometry
+FROM
+    county_2010census_dp1;
+"""
+    county_gdf = gpd.read_postgis(
+        sql,
+        con=dp1_engine,
+        geom_col="geometry",
+        crs="EPSG:4326",
     )
 
-    pudl_engine = sa.create_engine(PudlPaths().pudl_db)
-    # Load the US Census DP1 county data:
-    county_gdf = pudl.etl.defs.load_asset_value(AssetKey("county_censusdp1"))
-
-    kwargs_dicts = [
-        {"entity_type": "util", "limit_by_state": False},
-        {"entity_type": "util", "limit_by_state": True},
-        {"entity_type": "ba", "limit_by_state": True},
-        {"entity_type": "ba", "limit_by_state": False},
-    ]
```

Comment:
Rather than hard-coding that the script runs through all combinations of params, let the user select on the command line.